### PR TITLE
refactor(inlinealert)!: migrate core tokens

### PR DESCRIPTION
**Pull request by yosevu**
## Description

- Removed Help variant since it wasn't in the design
- Renamed success variant to positive to match design
- Renamed error variant to notice to match design
- Reordered variants
- Added WHCM styles

Does not include
- Typography components
- Updated min-width token

## How and where has this been tested?
- Chrome and Safari
- WHCM

## Screenshots

Screen Shot 2022-09-13 at 2 50 12 PM
Screen Shot 2022-09-13 at 2 45 55 PM
Screen Shot 2022-09-13 at 2 45 48 PM
Screen Shot 2022-09-13 at 2 44 02 PM
Screen Shot 2022-09-13 at 2 43 47 PM
Screen Shot 2022-09-13 at 2 43 38 PM

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.

- [ ] This pull request is ready to merge.

### CODE REVIEW

File: components/inlinealert/index.css [hunk 1]
```diff
@@ -10,81 +10,121 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-  --spectrum-inlinealert-neutral-title-height: 38px;
-  --spectrum-inlinealert-neutral-corner-radius: 4px;
-
-  /* DNA */
-  --spectrum-inlinealert-neutral-border-width: 2px;
+  /* Static properties */
+  --spectrum-inlinealert-font-size: 14px;
+  --spectrum-inlinealert-font-weight: 700;
+  --spectrum-inlinealert-font-style: normal;
+  --spectrum-inlinealert-line-height: 14px;
```
Comment (yosevu) [inline components/inlinealert/index.css:18]: There weren't tokens for these properties in the design.

File: components/inlinealert/index.css [hunk 2]
```diff
@@ -10,81 +10,121 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-  font-style: normal;
-  line-height: 14px;
-  text-transform: none;
+  color: var(--mod-inlinealert-header-and-content-color, var(--spectrum-inlinealert-header-and-content-color));
 }
 
 .spectrum-InLineAlert-content {
   display: block;
-  margin: var(--spectrum-inlinealert-neutral-content-margin-top) 0 0 0;
+  margin: var(--mod-inlinealert-spacing-header-content-button, var(--spectrum-inlinealert-spacing-header-content-button)) 0 0 0;
```
Comment (lunarfusion) [inline components/inlinealert/index.css:82]: Should we change this to margin-block-start and get rid of the  '0 0 0'?

File: components/inlinealert/index.css [hunk 3]
```diff
@@ -10,81 +10,121 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
-
-  /* DNA */
-  --spectrum-inlinealert-neutral-border-width: 2px;
+  /* Static properties */
+  --spectrum-inlinealert-font-size: 14px;
+  --spectrum-inlinealert-font-weight: 700;
+  --spectrum-inlinealert-font-style: normal;
+  --spectrum-inlinealert-line-height: 14px;
+
+  /* Size*/
```
Comment (lunarfusion) [inline components/inlinealert/index.css:20]: ✨ Custom properties are so nicely organized!
Nit: Line 20 and 27 lack a space before their closing */

File: components/inlinealert/package.json [hunk 4]
```diff
@@ -17,13 +17,13 @@
   },
   "peerDependencies": {
     "@spectrum-css/icon": "^3.0.21",
-    "@spectrum-css/vars": "^8.0.0"
+    "@spectrum-css/tokens": "^1.0.7"
   },
   "devDependencies": {
     "@spectrum-css/button": "^6.0.13",
     "@spectrum-css/component-builder": "^3.2.0",
```
Comment (pfulton) [inline components/inlinealert/package.json:24]: Can you remove this dependency, and instead use: `"@spectrum-css/component-builder-simple": "^1.0.0"`?

File: components/inlinealert/index.css (hunk 1 above)
Comment (pfulton) [reply components/inlinealert/index.css:18]: Would you mind following-up with design on this, @yosevu?

**Review by pfulton — CHANGES_REQUESTED** [review]
This is looking good, @yosevu - I left some comments here that we should address.

**Comment by yosevu** [comment]
@pfulton I had paused work on this while waiting for typography so I didn't finish adding all the token fallbacks. I have added them all now.

**Review by pfulton — APPROVED** [review]

File: components/inlinealert/index.css [hunk 5]
```diff
@@ -38,9 +38,11 @@ governing permissions and limitations under the License.
...
 
 @media (forced-colors: active) {
-  --highcontrast-inlinealert-background-color: Background;
-  --highcontrast-inlinealert-header-and-content-color: CanvasText;
-  --highcontrast-inlinealert-border-and-icon-color: ButtonBorder;
+  .spectrum-InLineAlert {
+    --highcontrast-inlinealert-background-color: Background;
+    --highcontrast-inlinealert-header-and-content-color: CanvasText;
+    --highcontrast-inlinealert-border-and-icon-color: ButtonBorder;
+  }
```
Comment (pfulton) [inline components/inlinealert/index.css:45]: @yosevu I missed this in my previous review, so I made the change myself. I think we should try to stick with scoping the `highcontrast` custom properties to the component's selector. Previously, (without the selector) these custom properties were just being defined without any selector scope.

**Comment by pfulton** [comment]
All right, this is rebased against the latest stuff in `main` and the beta has been published @yosevu:

`@spectrum-css/inlinealert@5.0.0-beta.0`

File: components/inlinealert/index.css (hunk 1 above)
Comment (yosevu) [reply components/inlinealert/index.css]: Resolved:

- Removed font tokens which will be handled in Typography Heading
- min-width token will be updated

File: components/inlinealert/index.css [hunk 6]
```diff
@@ -10,81 +10,121 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
   box-sizing: border-box;
-  min-inline-size: var(--spectrum-inlinealert-neutral-min-width);
-  min-block-size: var(--spectrum-inlinealert-neutral-title-height);
-  margin-block: 8px;
-  margin-inline: 0;
-  padding-block: var(--spectrum-inlinealert-neutral-padding-y);
-  padding-inline: var(--spectrum-inlinealert-neutral-padding-x);
-
-  border-width: var(--spectrum-inlinealert-neutral-border-width);
+  min-inline-size: var(--mod-inlinealert-min-inline-size, var(--spectrum-inlinealert-min-inline-size));
```
[the remark this answers is not in the document]
Comment (yosevu) [reply components/inlinealert/index.css:46]: The min-width token will be updated.

**Review by pfulton — CHANGES_REQUESTED** [review]
Would you mind taking a look at this in Firefox? The borders are looking a little funky:

Screen Shot 2022-10-05 at 10 27 06 AM

I _think_ you might need to also define `border-inline-width` in addition to your existing `border-block-width` (at least, that's what I tried in my browser inspector and it seemed to work ok?)

**Review by pfulton — APPROVED** [review]
This looks good now, @yosevu! I also rebased this against `main` and bumped the beta version number. It's been released at: `@spectrum-css/inlinealert@5.0.0-beta.1`

**Pull request merged**